deepface/modules/recognition.py

```diff
@@ -304,11 +306,11 @@ def __list_images(path: str) -> list[str]:
         images (list): list of exact image paths
     """
     images = []
-    for r, _, f in os.walk(path):
-        for file in f:
-            if file.lower().endswith((".jpg", ".jpeg", ".png")):
-                exact_path = os.path.join(r, file)
-                images.append(exact_path)
+    pattern = re.compile(r".*\.(jpg|jpeg|png)$", re.IGNORECASE)
```

Comment:
prefer not to use regex, would you please restore this block to its old version.

Reply:
Done ... just fyi the regex version shaves about 20milliseconds from every loop.